packages/flame/lib/src/game/notifying_vector2.dart

```diff
@@ -13,7 +13,15 @@ import 'package:vector_math/vector_math_64.dart';
 /// Direct modification of this vector's [storage] is not allowed.
 ///
 class NotifyingVector2 extends Vector2 with ChangeNotifier {
-  NotifyingVector2() : super.zero();
+  factory NotifyingVector2(double x, double y) =>
+      NotifyingVector2.zero()..setValues(x, y);
```

Comment:
This doesn't need to be a factory right? It could just be a constructor that invokes super?

Reply:
I'm not particularly sure what's going on here, but if I remove `factory` keyword it gives an error
```
The constructor 'Vector2 Vector2(double x, double y)' is a factory constructor, but must be a generative constructor to be a valid superinitializer.
```
Also, this is how they've set up constructors in the base `Vector2` class, so I just followed the lead.

Reply:
I see, let's go with that then.